ros2topic/ros2topic/verb/pub.py

```diff
@@ -147,14 +153,19 @@ def publisher(
     pub = node.create_publisher(msg_module, topic_name, qos_profile)
 
     times_since_last_log = 0
+    total_wait_time = 0
     while pub.get_subscription_count() < wait_matching_subscriptions:
         # Print a message reporting we're waiting each 1s, check condition each 100ms.
         if not times_since_last_log:
             print(
                 f'Waiting for at least {wait_matching_subscriptions} matching subscription(s)...')
+        if max_wait_time is not None and max_wait_time <= total_wait_time:
+            return 'Timed out waiting for subscribers'       
```

Comment:
it would be helpful to print how many subscriptions we are missing when it times out?

Reply:
See c6dda6e